File: plugin/src/org/jetbrains/haskell/fileType/HiFileType.java
package org.jetbrains.haskell.fileType;

import com.intellij.openapi.fileTypes.FileType;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class HiFileType implements FileType {

    public static final HiFileType INSTANCE = new HiFileType();


    @NotNull
    public String getName() {
        return "Haskell interface";
    }

    @NotNull
    public String getDescription() {
        return "Haskell interface file";
    }

    @NotNull
    public String getDefaultExtension() {
        return "hi";
    }

    public Icon getIcon() {
        return null;
    }

    public boolean isBinary() {
        return true;
    }

    public boolean isReadOnly() {
        return true;
    }

    public String getCharset(@NotNull VirtualFile file, byte[] content) {
        return null;
    }

}

File: plugin/src/org/jetbrains/haskell/fileType/HiFileType.kt
package org.jetbrains.haskell.fileType

import com.intellij.openapi.fileTypes.FileType
import com.intellij.openapi.vfs.VirtualFile
import javax.swing.*

public class HiFileType() : FileType {


    override fun getName(): String {
        return "Haskell interface"
    }

    override fun getDescription(): String {
        return "Haskell interface file"
    }

    override fun getDefaultExtension(): String {
        return "hi"
    }

    override fun getIcon(): Icon? {
        return null
    }

    override fun isBinary(): Boolean {
        return true
    }

    override fun isReadOnly(): Boolean {
        return true
    }

    override fun getCharset(file: VirtualFile, content: ByteArray?): String? {
        return null
    }


    class object {
        public val INSTANCE: HiFileType = HiFileType()
    }
}